### PR TITLE
fix: require min two cosigners, validate duplicated keys, fix typography

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA -[LW-10994](https://input-output.atlassian.net/browse/LW-10994)
- [x] JIRA -[LW-10995](https://input-output.atlassian.net/browse/LW-10995)
- [x] JIRA -[LW-10997](https://input-output.atlassian.net/browse/LW-10997)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

[LW-10994]: https://input-output.atlassian.net/browse/LW-10994?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LW-10995]: https://input-output.atlassian.net/browse/LW-10995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[LW-10997]: https://input-output.atlassian.net/browse/LW-10997?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by pczeglik-iohk** [comment]
# Allure Report
[`allure-report-publisher`](https://github.com/andrcuns/allure-report-publisher) generated test report!



**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/smoke/linux/chrome/6328/10047381642/index.html) for [b27c1250](https://github.com/input-output-hk/lace/pull/1315/commits/b27c12504d99b97a8159d2d48af48fd81ce2b963)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 32     | 0      | 0       | 0     | 32    | ✅     |

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSignerInput.tsx [hunk 1]
```diff
@@ -39,6 +39,9 @@ const parseError = (error: CoSignerError | undefined, t: TFunction): Partial<Rec
   if (error.sharedWalletKey === CoSignerErrorKeys.Invalid) {
     keyErrorMessage = t('sharedWallets.addSharedWallet.addCosigners.keyInputError.invalid');
   }
+  if (error.sharedWalletKey === CoSignerErrorKeys.Duplicated) {
```
Comment (danielmain) [inline packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSignerInput.tsx:42]: [OPTIONAL] What about a switch?

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx [hunk 2]
```diff
@@ -27,7 +29,8 @@ export const AddCoSigners = ({
   coSignersDirty,
 }: Props): JSX.Element => {
   const { t } = useTranslation();
-  const atLeastOneValidCoSigner = coSigners.some((c) => c.sharedWalletKey && c.name) && errors.length === 0;
+  const atLeastTwoValidCoSigners =
```
Comment (danielmain) [inline packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx:32]: 🙄 the name of the variable says `Two`, but the MIN_NUMBER_OF_COSIGNERS can potentially be 2,3,etc.

Rename it to `hasAtLeastMinValidCoSigners`

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx [hunk 3]
```diff
@@ -37,7 +40,7 @@ export const AddCoSigners = ({
       timelineCurrentStep={SharedWalletCreationStep.CoSigners}
       onBack={onBack}
       onNext={onNext}
-      isNextEnabled={atLeastOneValidCoSigner}
+      isNextEnabled={atLeastTwoValidCoSigners}
```
Comment (danielmain) [inline packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx:43]: `hasAtLeastMinValidCoSigners`

**Review by danielmain — CHANGES_REQUESTED** [review]
Left one optional change and other suggestions

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx [hunk 4]
```diff
@@ -9,6 +9,8 @@ import { AddCoSignerInput } from './AddCoSignerInput';
 import styles from './AddCoSigners.module.scss';
 import { CoSigner, CoSignerDirty, CoSignerError } from './type';
 
+const MIN_NUMBER_OF_COSIGNERS = 2;
```
[the remark this answers is not in the document]
Comment (vetalcore) [reply packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx]: done here 521b6f1ab78e3159fb7d48c7c86ec3e909a66e5a

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx (hunk 2 above)
Comment (vetalcore) [reply packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx]: done here 521b6f1ab78e3159fb7d48c7c86ec3e909a66e5a

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx (hunk 3 above)
Comment (vetalcore) [reply packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx]: done here 521b6f1ab78e3159fb7d48c7c86ec3e909a66e5a

**Review by shawnbusuttil — APPROVED** [review]

File: apps/browser-extension-wallet/.env.defaults [hunk 5]
```diff
@@ -93,3 +93,4 @@ MIDNIGHT_EVENT_BANNER_REMINDER_TIME=129600000
 
 # Shared Wallet
 SHARED_WALLET_TX_VALIDITY_INTERVAL=24
+MIN_NUMBER_OF_COSIGNERS=2
```
Comment (greatertomi) [inline apps/browser-extension-wallet/.env.defaults:96]: Why do we need this?

Comment (szymonmaslowski) [reply apps/browser-extension-wallet/.env.defaults:96]: I share the same question 😉 I think this should not be a subject for configuration, it is rather a nature of the feature

**Review by szymonmaslowski — APPROVED** [review]
GJ! Just one think about having constant.

File: packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (szymonmaslowski) [reply packages/core/src/shared-wallets/add-shared-wallet/creation-flow/AddCoSigners/AddCoSigners.tsx]: @danielmain I don't think this should be placed in the env file. This is a constant determined by the nature of the feature, right? It should not be configurable I guess...

File: apps/browser-extension-wallet/.env.defaults (hunk 5 above)
Comment (vetalcore) [reply apps/browser-extension-wallet/.env.defaults:96]: @danielmain wdyt?

**Review by danielmain — APPROVED** [review]
Awesome job Vit!

**Pull request merged**